script.js

```diff
@@ -0,0 +1,105 @@
+let sourceTextBox = document.getElementById('source')
+let preview = document.getElementById('preview')
+let markupButton = document.getElementById('render')
+let boldButton = document.getElementById('bold')
+let italicizeButton = document.getElementById('italicize')
+let selection = ""
+
+function markdownText() {
+    let sourceText = sourceTextBox.value
+    preview.innerHTML = marked(sourceText)
+}
+function selectedText(event) {
+    selection = event.target.value.substring(event.target.selectionStart, event.target.selectionEnd);
```

Comment:
i think I found a bug ...  `selection` is global, so if this function isn't called prior to `boldText` being called, `selection` will be undefined (or have an out-of-date value)... better use `let` and also call it from a different event handler later on. I'm not expressing myself very clearly so check in with me in person and I'll better explain what I mean.

Reply:
Imagine if instead of being an event handler, this function takes in an *element* as its parameter, and returns a string...

Reply:
or alternately, since it's an event handler, it should return nothing